mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,801 @@
+        # gt_list member: [label, start_frame, end_frame]
+        # proposal_list member: [label, best_iou, overlap_self,
+        #                        start_frame, end_frame]
+        for proposal_info in proposal_infos:
+            if self.data_prefix is not None:
+                frame_dir = osp.join(self.data_prefix, proposal_info[0])
+            num_frames = int(proposal_info[1])
+            # gts:start, end, num_frames, class_label, tIoU=1
+            gts = []
+            for x in proposal_info[2]:
+                if int(x[2]) > int(x[1]):
+                    ssn_instance = SSNInstance(
+                        int(x[1]),
+                        int(x[2]),
+                        num_frames,
+                        label=int(x[0]),
+                        best_iou=1.0)
+                    gts.append(ssn_instance)
+            gts = list(filter(lambda x: x.start_frame < num_frames, gts))
+            # proposals:start, end, num_frames, class_label
+            # tIoU=best_iou, overlap_self
+            proposals = []
+            for x in proposal_info[3]:
+                if int(x[4]) > int(x[3]):
+                    ssn_instance = SSNInstance(
+                        int(x[3]),
+                        int(x[4]),
+                        num_frames,
+                        label=int(x[0]),
+                        best_iou=float(x[1]),
+                        overlap_self=float(x[2]))
+                    proposals.append(ssn_instance)
+            proposals = list(
+                filter(lambda x: x.start_frame < num_frames, proposals))
+            video_infos.append(
+                dict(
+                    frame_dir=frame_dir,
+                    video_id=proposal_info[0],
+                    total_frames=num_frames,
+                    gts=gts,
+                    proposals=proposals))
+        return video_infos
+
+    def evaluate(self,
+                 results,
+                 metrics='mAP',
+                 eval_dataset='thumos14',
+                 **kwargs):
+        """Evaluation in SSN proposal dataset.
+
+        Args:
+            results (list[dict]): Output results.
+            metrics (str | sequence[str]): Metrics to be performed.
+                Defaults: 'mAP'.
+            eval_dataset (str): Dataset to be evaluated.
+
+        Returns:
+            dict: Evaluation results for evaluation metrics.
+        """
+        if not isinstance(results, list):
+            raise TypeError(f'results must be a list, but got {type(results)}')
+        assert len(results) == len(self), (
+            f'The length of results is not equal to the dataset len: '
+            f'{len(results)} != {len(self)}')
+
+        metrics = metrics if isinstance(metrics, (list, tuple)) else [metrics]
+        allowed_metrics = ['mAP']
+        for metric in metrics:
+            if metric not in allowed_metrics:
+                raise KeyError(f'metric {metric} is not supported')
+
```

Comment:
```
allowed_eval_dataset = ['thumos14']
if eval_dataset not in allowed_eval_dataset:
    raise NotImplementedError(f'metric {metric} is not supported')
```

Reply:
BTW, Do the codes only support 'thumos14' ?

Reply:
Yes, the original codes only support thumos14, we can add 'activitynet' in the future.